diff --git rules/android_sdk_repository/empty.template.bzl rules/android_sdk_repository/empty.template.bzl
index 272e666..b75b7fd 100644
--- rules/android_sdk_repository/empty.template.bzl
+++ rules/android_sdk_repository/empty.template.bzl
@@ -15,6 +15,8 @@
 This module adds no-op repository rules when the Android SDK is not installed.
 """
 
+load("@bazel_skylib//rules:common_settings.bzl", "bool_setting")
+
 package(default_visibility = ["//visibility:public"])
 
 # android_sdk_repository was used without a valid Android SDK being set.
@@ -25,13 +27,9 @@ package(default_visibility = ["//visibility:public"])
 # The toolchain type used to distinguish Android SDK toolchains.
 toolchain_type(name = "sdk_toolchain_type")
 
-config_feature_flag(
+bool_setting(
     name = "true",
-    allowed_values = [
-        "true",
-        "false",
-    ],
-    default_value = "true",
+    build_setting_default = True,
 )
 
 config_setting(
diff --git rules/android_sdk_repository/template.bzl rules/android_sdk_repository/template.bzl
index 0181a49..4ea3414 100644
--- rules/android_sdk_repository/template.bzl
+++ rules/android_sdk_repository/template.bzl
@@ -15,6 +15,7 @@
 This module contains repository rules and helpers needed to configure the Android SDK for Bazel.
 """
 
+load("@bazel_skylib//rules:common_settings.bzl", "bool_setting")
 load(
     ":helper.bzl",
     "create_android_sdk_rules",
@@ -26,13 +27,9 @@ package(default_visibility = ["//visibility:public"])
 # The toolchain type used to distinguish Android SDK toolchains.
 toolchain_type(name = "sdk_toolchain_type")
 
-config_feature_flag(
+bool_setting(
     name = "true",
-    allowed_values = [
-        "true",
-        "false",
-    ],
-    default_value = "true",
+    build_setting_default = True,
 )
 
 config_setting(
