diff --git go/private/repositories.bzl go/private/repositories.bzl
index 8560d67f..39a07ddd 100644
--- go/private/repositories.bzl
+++ go/private/repositories.bzl
@@ -199,6 +199,7 @@ def go_rules_dependencies(is_rules_go = False):
         patches = [
             # releaser:patch-cmd gazelle -repo_root . -go_prefix github.com/golang/protobuf -go_naming_convention import_alias -proto disable_global
             "@io_bazel_rules_go//third_party:com_github_golang_protobuf-gazelle.patch",
+            "@com_github_buildbarn_bb_storage//:patches/com_github_golang_protobuf/service-registrar.diff",
         ],
         patch_args = ["-p1"],
     )
