diff --git uploader.go uploader.go
index 26fbf1cb..e2ebe5be 100644
--- uploader.go
+++ uploader.go
@@ -185,7 +185,7 @@ type collectorEndpointConfig struct {
 	password string
 
 	// httpClient to be used to make requests to the collector endpoint.
-	httpClient *http.Client
+	httpClient HTTPClient
 }
 
 type collectorEndpointOptionFunc func(*collectorEndpointConfig)
@@ -225,8 +225,12 @@ func WithPassword(password string) CollectorEndpointOption {
 	})
 }
 
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
 // WithHTTPClient sets the http client to be used to make request to the collector endpoint.
-func WithHTTPClient(client *http.Client) CollectorEndpointOption {
+func WithHTTPClient(client HTTPClient) CollectorEndpointOption {
 	return collectorEndpointOptionFunc(func(o *collectorEndpointConfig) {
 		o.httpClient = client
 	})
@@ -266,7 +270,7 @@ type collectorUploader struct {
 	endpoint   string
 	username   string
 	password   string
-	httpClient *http.Client
+	httpClient HTTPClient
 }
 
 var _ batchUploader = (*collectorUploader)(nil)
