diff --git cc/private/link/cpp_link_action.bzl cc/private/link/cpp_link_action.bzl
index 764c626..7b47923 100644
--- cc/private/link/cpp_link_action.bzl
+++ cc/private/link/cpp_link_action.bzl
@@ -124,7 +124,7 @@ def link_action(
         fail("Expected action_config for '%s' to be configured" % link_type.action_name)
 
     object_files = compilation_outputs.pic_objects if use_pic else compilation_outputs.objects
-    object_files = object_files + additional_object_files
+    object_files = list(object_files) + additional_object_files
 
     propeller_optimize_info = getattr(cc_toolchain._fdo_context, "propeller_optimize_info", None)
     if not cc_toolchain._is_tool_configuration and propeller_optimize_info and \
diff --git cc/private/link/create_library_to_link.bzl cc/private/link/create_library_to_link.bzl
index f2b85bd..29d21f0 100644
--- cc/private/link/create_library_to_link.bzl
+++ cc/private/link/create_library_to_link.bzl
@@ -102,8 +102,8 @@ def make_library_to_link(
         pic_objects = _cc_internal.freeze(pic_objects),
         alwayslink = alwayslink,
         # LTO data duplication is forced by public APIs
-        lto_bitcode_files = _cc_internal.freeze(_lto_compilation_context.lto_bitcode_inputs.keys() if _lto_compilation_context else []),
-        pic_lto_bitcode_files = _cc_internal.freeze(_pic_lto_compilation_context.lto_bitcode_inputs.keys() if _pic_lto_compilation_context else []),
+        lto_bitcode_files = _cc_internal.freeze(dict(_lto_compilation_context.lto_bitcode_inputs).keys() if _lto_compilation_context else []),
+        pic_lto_bitcode_files = _cc_internal.freeze(dict(_pic_lto_compilation_context.lto_bitcode_inputs).keys() if _pic_lto_compilation_context else []),
         _library_identifier = _library_identifier,
         _contains_objects = _contains_objects,
         _must_keep_debug = _must_keep_debug,
