diff --git build/bazel/remote/execution/v2/BUILD build/bazel/remote/execution/v2/BUILD
index 01f415b..eb3b3f2 100644
--- build/bazel/remote/execution/v2/BUILD
+++ build/bazel/remote/execution/v2/BUILD
@@ -1,7 +1,9 @@
-package(default_visibility = ["//visibility:public"])
-
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")
 load("@rules_proto//proto:defs.bzl", "proto_library")
 
+package(default_visibility = ["//visibility:public"])
+
 licenses(["notice"])
 
 proto_library(
@@ -13,29 +15,28 @@ proto_library(
         "@com_google_protobuf//:duration_proto",
         "@com_google_protobuf//:timestamp_proto",
         "@com_google_protobuf//:wrappers_proto",
-        "@googleapis//:google_api_annotations_proto",
-        "@googleapis//:google_api_http_proto",
-        "@googleapis//:google_longrunning_operations_proto",
-        "@googleapis//:google_rpc_status_proto",
+        "@googleapis//google/api:annotations_proto",
+        "@googleapis//google/api:http_proto",
+        "@googleapis//google/longrunning:operations_proto",
+        "@googleapis//google/rpc:status_proto",
     ],
 )
 
-alias(
-    name = "remote_execution_java_proto",
-    actual = "//build/bazel/remote/execution/v2/java:remote_execution_java_proto",
-)
-
-alias(
-    name = "remote_execution_cc_proto",
-    actual = "//build/bazel/remote/execution/v2/cc:remote_execution_cc_grpc",
-)
-
-alias(
+go_proto_library(
     name = "remote_execution_go_proto",
-    actual = "//build/bazel/remote/execution/v2/go:remote_execution_go_proto",
+    compilers = ["@io_bazel_rules_go//proto:go_grpc"],
+    importpath = "github.com/bazelbuild/remote-apis/build/bazel/remote/execution/v2",
+    proto = ":remote_execution_proto",
+    deps = [
+        "//build/bazel/semver",
+        "@com_google_cloud_go_longrunning//autogen/longrunningpb",
+        "@org_golang_google_genproto_googleapis_api//annotations",
+        "@org_golang_google_genproto_googleapis_rpc//status",
+    ],
 )
 
-alias(
-    name = "go_default_library",
-    actual = "//build/bazel/remote/execution/v2/go:go_default_library",
+go_library(
+    name = "execution",
+    embed = [":remote_execution_go_proto"],
+    importpath = "github.com/bazelbuild/remote-apis/build/bazel/remote/execution/v2",
 )
diff --git build/bazel/semver/BUILD build/bazel/semver/BUILD
index 9663c94..f11d489 100644
--- build/bazel/semver/BUILD
+++ build/bazel/semver/BUILD
@@ -1,5 +1,7 @@
 package(default_visibility = ["//visibility:public"])
 
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")
 load("@rules_proto//proto:defs.bzl", "proto_library")
 
 licenses(["notice"])
@@ -9,22 +11,14 @@ proto_library(
     srcs = ["semver.proto"],
 )
 
-alias(
-    name = "semver_java_proto",
-    actual = "//build/bazel/semver/java:semver_java_proto",
-)
-
-alias(
-    name = "semver_cc_proto",
-    actual = "//build/bazel/semver/cc:semver_cc_proto",
-)
-
-alias(
+go_proto_library(
     name = "semver_go_proto",
-    actual = "//build/bazel/semver/go:semver_go_proto",
+    importpath = "github.com/bazelbuild/remote-apis/build/bazel/semver",
+    proto = ":semver_proto",
 )
 
-alias(
-    name = "go_default_library",
-    actual = "//build/bazel/semver/go:go_default_library",
+go_library(
+    name = "semver",
+    embed = [":semver_go_proto"],
+    importpath = "github.com/bazelbuild/remote-apis/build/bazel/semver",
 )
