diff --git python/private/enum.bzl python/private/enum.bzl
index 4d0fb106..b9ced015 100644
--- python/private/enum.bzl
+++ python/private/enum.bzl
@@ -39,13 +39,13 @@ def enum(methods = {}, **kwargs):
 
     for name, unbound_method in methods.items():
         # buildifier: disable=uninitialized
-        kwargs[name] = lambda *a, **k: unbound_method(self, *a, **k)
+        kwargs[name] = lambda *a, **k: unbound_method(members, *a, **k)
 
     self = struct(__members__ = members, **kwargs)
     return self
 
-def _FlagEnum_flag_values(self):
-    return sorted(self.__members__.values())
+def _FlagEnum_flag_values(members):
+    return sorted(members.values())
 
 def FlagEnum(**kwargs):
     """Define an enum specialized for flags.
