diff --git MODULE.bazel MODULE.bazel
index 9992b24a8..713e8ca99 100644
--- MODULE.bazel
+++ MODULE.bazel
@@ -32,7 +32,7 @@ use_repo(
     "rules_rust_tinyjson",
 )
 
-cargo_internal_deps = use_extension("//cargo/private:internal_extensions.bzl", "i")
+cargo_internal_deps = use_extension("//cargo/private:internal_extensions.bzl", "j")
 use_repo(
     cargo_internal_deps,
     "rrcti",
diff --git cargo/private/internal_extensions.bzl cargo/private/internal_extensions.bzl
index a025f53e7..c0a70b188 100644
--- cargo/private/internal_extensions.bzl
+++ cargo/private/internal_extensions.bzl
@@ -26,7 +26,7 @@ def _internal_deps_impl(module_ctx):
 # This is named a single character to reduce the size of path names when running build scripts, to reduce the chance
 # of hitting the 260 character windows path name limit.
 # TODO: https://github.com/bazelbuild/rules_rust/issues/1120
-i = module_extension(
+j = module_extension(
     doc = "Dependencies for rules_rust",
     implementation = _internal_deps_impl,
 )
