diff --git internal/gengogrpc/grpc.go internal/gengogrpc/grpc.go
index fd2f51d..d5f1c21 100644
--- internal/gengogrpc/grpc.go
+++ internal/gengogrpc/grpc.go
@@ -153,7 +153,7 @@ func genService(gen *protogen.Plugin, file *protogen.File, g *protogen.Generated
 		g.P(deprecationComment)
 	}
 	serviceDescVar := "_" + service.GoName + "_serviceDesc"
-	g.P("func Register", service.GoName, "Server(s *", grpcPackage.Ident("Server"), ", srv ", serverType, ") {")
+	g.P("func Register", service.GoName, "Server(s ", grpcPackage.Ident("ServiceRegistrar"), ", srv ", serverType, ") {")
 	g.P("s.RegisterService(&", serviceDescVar, `, srv)`)
 	g.P("}")
 	g.P()
diff --git protoc-gen-go/grpc/grpc.go protoc-gen-go/grpc/grpc.go
index 40cba16..f2fd6ba 100644
--- protoc-gen-go/grpc/grpc.go
+++ protoc-gen-go/grpc/grpc.go
@@ -212,7 +212,7 @@ func (g *grpc) generateService(file *generator.FileDescriptor, service *pb.Servi
 	if deprecated {
 		g.P(deprecationComment)
 	}
-	g.P("func Register", servName, "Server(s *", grpcPkg, ".Server, srv ", serverType, ") {")
+	g.P("func Register", servName, "Server(s ", grpcPkg, ".ServiceRegistrar, srv ", serverType, ") {")
 	g.P("s.RegisterService(&", serviceDescVar, `, srv)`)
 	g.P("}")
 	g.P()
