diff --git astgen/BUILD.bazel astgen/BUILD.bazel
index 2d0a4b5..b326c13 100644
--- astgen/BUILD.bazel
+++ astgen/BUILD.bazel
@@ -1,13 +1,5 @@
 load("@io_bazel_rules_go//go:def.bzl", "go_library")
 
-genrule(
-    name = "dumpstdlibast",
-    srcs = ["@cpp_jsonnet//stdlib"],
-    outs = ["stdast.go"],
-    cmd = "./$(location //cmd/dumpstdlibast) \"$<\" > \"$@\"",
-    tools = ["//cmd/dumpstdlibast"],
-)
-
 go_library(
     name = "go_default_library",
     srcs = ["stdast.go"],
