diff --git go/private/context.bzl go/private/context.bzl
index 4b08b920..f5cd735d 100644
--- go/private/context.bzl
+++ go/private/context.bzl
@@ -413,7 +413,7 @@ def _infer_importpath(ctx, embeds, importpath, importmap):
 def matches_scope(label, scope):
     if scope == "all":
         return True
-    if scope.workspace_name != label.workspace_name:
+    if scope.repo_name != label.repo_name:
         return False
     if scope.name == "__pkg__":
         return scope.package == label.package
diff --git go/private/extensions.bzl go/private/extensions.bzl
index a5c8fc9b..f3c8f154 100644
--- go/private/extensions.bzl
+++ go/private/extensions.bzl
@@ -129,7 +129,7 @@ def _go_sdk_impl(ctx):
         for scope in nogo_tag.includes + nogo_tag.excludes:
             # Validate that the scope references a valid, visible repository.
             # buildifier: disable=no-effect
-            scope.workspace_name
+            scope.repo_name
             if scope.name != "__pkg__" and scope.name != "__subpackages__":
                 fail(
                     "go_sdk.nogo: all entries in includes and excludes must end with ':__pkg__' or ':__subpackages__', got '{}' in".format(scope.name),
diff --git go/private/rules/test.bzl go/private/rules/test.bzl
index 066c6892..4859c53e 100644
--- go/private/rules/test.bzl
+++ go/private/rules/test.bzl
@@ -107,12 +107,12 @@ def _go_test_impl(ctx):
 
     # now generate the main function
     repo_relative_rundir = ctx.attr.rundir or ctx.label.package or "."
-    if ctx.label.workspace_name:
-        # The test is contained in an external repository (Label.workspace_name is always the empty
+    if ctx.label.repo_name:
+        # The test is contained in an external repository (Label.repo_name is always the empty
         # string for the main repository, which is the canonical repository name of this repo).
         # The test runner cd's into the directory corresponding to the main repository, so walk up
         # and then down.
-        run_dir = "../" + ctx.label.workspace_name + "/" + repo_relative_rundir
+        run_dir = "../" + ctx.label.repo_name + "/" + repo_relative_rundir
     else:
         run_dir = repo_relative_rundir
 
