diff --git go/private/context.bzl go/private/context.bzl
index 1eb69a5e..aa7d8f3c 100644
--- go/private/context.bzl
+++ go/private/context.bzl
@@ -419,7 +419,7 @@ def _infer_importpath(ctx, attr):
 def matches_scope(label, scope):
     if scope == "all":
         return True
-    if scope.workspace_name != label.workspace_name:
+    if scope.repo_name != label.repo_name:
         return False
     if scope.name == "__pkg__":
         return scope.package == label.package
