diff --git platform_defs/BUILD platform_defs/BUILD
index 930d773..4d9a3c4 100644
--- platform_defs/BUILD
+++ platform_defs/BUILD
@@ -1,10 +1,10 @@
-load("@bazel_tools//tools/python:toolchain.bzl", "py_runtime_pair")
+load("@@rules_python+//python:defs.bzl", "py_runtime", "py_runtime_pair")
 
 py_runtime(
     name = "python3",
     interpreter_path = "/usr/bin/python3",
-    stub_shebang = "#!/usr/bin/env python3",
     python_version = "PY3",
+    stub_shebang = "#!/usr/bin/env python3",
 )
 
 py_runtime_pair(
@@ -16,6 +16,5 @@ py_runtime_pair(
 toolchain(
     name = "default_python3_toolchain",
     toolchain = ":just_python3",
-    toolchain_type = "@bazel_tools//tools/python:toolchain_type",
+    toolchain_type = "@@rules_python+//python:toolchain_type",
 )
-
