diff --git js/private/js_binary.bzl js/private/js_binary.bzl
index 1d4044c3..eda0ea94 100644
--- js/private/js_binary.bzl
+++ js/private/js_binary.bzl
@@ -360,14 +360,14 @@ def _bash_launcher(ctx, nodeinfo, entry_point_path, log_prefix_rule_set, log_pre
 
     # Add common and useful make variables to the environment
     makevars = {
-        "JS_BINARY__BINDIR": "$(BINDIR)",
-        "JS_BINARY__COMPILATION_MODE": "$(COMPILATION_MODE)",
-        "JS_BINARY__TARGET_CPU": "$(TARGET_CPU)",
+        "JS_BINARY__BINDIR": "BINDIR",
+        "JS_BINARY__COMPILATION_MODE": "COMPILATION_MODE",
+        "JS_BINARY__TARGET_CPU": "TARGET_CPU",
     }
     for (key, value) in makevars.items():
         envs.append(_ENV_SET.format(
             var = key,
-            value = ctx.expand_make_variables("env", value, {}),
+            value = ctx.var[value],
         ))
 
     # Add rule context variables to the environment
