diff --git unix/zerrors_darwin_amd64.go unix/zerrors_darwin_amd64.go
index d73c465..d406964 100644
--- unix/zerrors_darwin_amd64.go
+++ unix/zerrors_darwin_amd64.go
@@ -1128,6 +1128,7 @@ const (
 	O_DSYNC                                 = 0x400000
 	O_EVTONLY                               = 0x8000
 	O_EXCL                                  = 0x800
+	O_EXEC                                  = 0x40000000
 	O_EXLOCK                                = 0x20
 	O_FSYNC                                 = 0x80
 	O_NDELAY                                = 0x4
@@ -1138,6 +1139,7 @@ const (
 	O_POPUP                                 = 0x80000000
 	O_RDONLY                                = 0x0
 	O_RDWR                                  = 0x2
+	O_SEARCH                                = 0x40100000
 	O_SHLOCK                                = 0x10
 	O_SYMLINK                               = 0x200000
 	O_SYNC                                  = 0x80
diff --git unix/zerrors_darwin_arm64.go unix/zerrors_darwin_arm64.go
index 4a55a40..c47c6e9 100644
--- unix/zerrors_darwin_arm64.go
+++ unix/zerrors_darwin_arm64.go
@@ -1128,6 +1128,7 @@ const (
 	O_DSYNC                                 = 0x400000
 	O_EVTONLY                               = 0x8000
 	O_EXCL                                  = 0x800
+	O_EXEC                                  = 0x40000000
 	O_EXLOCK                                = 0x20
 	O_FSYNC                                 = 0x80
 	O_NDELAY                                = 0x4
@@ -1138,6 +1139,7 @@ const (
 	O_POPUP                                 = 0x80000000
 	O_RDONLY                                = 0x0
 	O_RDWR                                  = 0x2
+	O_SEARCH                                = 0x40100000
 	O_SHLOCK                                = 0x10
 	O_SYMLINK                               = 0x200000
 	O_SYNC                                  = 0x80
