diff --git unix/mkerrors.sh unix/mkerrors.sh
index 3f670fa..6e6afca 100755
--- unix/mkerrors.sh
+++ unix/mkerrors.sh
@@ -563,6 +563,7 @@ ccflags="$@"
 		$2 ~ /^KEYCTL_/ ||
 		$2 ~ /^PERF_/ ||
 		$2 ~ /^SECCOMP_MODE_/ ||
+		$2 ~ /^SEEK_/ ||
 		$2 ~ /^SPLICE_/ ||
 		$2 ~ /^SYNC_FILE_RANGE_/ ||
 		$2 !~ /^AUDIT_RECORD_MAGIC/ &&
diff --git unix/zerrors_darwin_amd64.go unix/zerrors_darwin_amd64.go
index 991996b..5bb48ef 100644
--- unix/zerrors_darwin_amd64.go
+++ unix/zerrors_darwin_amd64.go
@@ -1262,6 +1262,11 @@ const (
 	SCM_RIGHTS                        = 0x1
 	SCM_TIMESTAMP                     = 0x2
 	SCM_TIMESTAMP_MONOTONIC           = 0x4
+	SEEK_CUR                          = 0x1
+	SEEK_DATA                         = 0x4
+	SEEK_END                          = 0x2
+	SEEK_HOLE                         = 0x3
+	SEEK_SET                          = 0x0
 	SHUT_RD                           = 0x0
 	SHUT_RDWR                         = 0x2
 	SHUT_WR                           = 0x1
diff --git unix/zerrors_darwin_arm64.go unix/zerrors_darwin_arm64.go
index e644eaf..11e5709 100644
--- unix/zerrors_darwin_arm64.go
+++ unix/zerrors_darwin_arm64.go
@@ -1262,6 +1262,11 @@ const (
 	SCM_RIGHTS                        = 0x1
 	SCM_TIMESTAMP                     = 0x2
 	SCM_TIMESTAMP_MONOTONIC           = 0x4
+	SEEK_CUR                          = 0x1
+	SEEK_DATA                         = 0x4
+	SEEK_END                          = 0x2
+	SEEK_HOLE                         = 0x3
+	SEEK_SET                          = 0x0
 	SHUT_RD                           = 0x0
 	SHUT_RDWR                         = 0x2
 	SHUT_WR                           = 0x1
diff --git unix/zerrors_freebsd_386.go unix/zerrors_freebsd_386.go
index 9c7c5e1..4409001 100644
--- unix/zerrors_freebsd_386.go
+++ unix/zerrors_freebsd_386.go
@@ -1297,6 +1297,11 @@ const (
 	SCM_RIGHTS                     = 0x1
 	SCM_TIMESTAMP                  = 0x2
 	SCM_TIME_INFO                  = 0x7
+	SEEK_CUR                       = 0x1
+	SEEK_DATA                      = 0x3
+	SEEK_END                       = 0x2
+	SEEK_HOLE                      = 0x4
+	SEEK_SET                       = 0x0
 	SHUT_RD                        = 0x0
 	SHUT_RDWR                      = 0x2
 	SHUT_WR                        = 0x1
diff --git unix/zerrors_freebsd_amd64.go unix/zerrors_freebsd_amd64.go
index b265abb..64520d3 100644
--- unix/zerrors_freebsd_amd64.go
+++ unix/zerrors_freebsd_amd64.go
@@ -1298,6 +1298,11 @@ const (
 	SCM_RIGHTS                     = 0x1
 	SCM_TIMESTAMP                  = 0x2
 	SCM_TIME_INFO                  = 0x7
+	SEEK_CUR                       = 0x1
+	SEEK_DATA                      = 0x3
+	SEEK_END                       = 0x2
+	SEEK_HOLE                      = 0x4
+	SEEK_SET                       = 0x0
 	SHUT_RD                        = 0x0
 	SHUT_RDWR                      = 0x2
 	SHUT_WR                        = 0x1
diff --git unix/zerrors_freebsd_arm.go unix/zerrors_freebsd_arm.go
index 3df99f2..99e9a0e 100644
--- unix/zerrors_freebsd_arm.go
+++ unix/zerrors_freebsd_arm.go
@@ -1276,6 +1276,11 @@ const (
 	SCM_CREDS                      = 0x3
 	SCM_RIGHTS                     = 0x1
 	SCM_TIMESTAMP                  = 0x2
+	SEEK_CUR                       = 0x1
+	SEEK_DATA                      = 0x3
+	SEEK_END                       = 0x2
+	SEEK_HOLE                      = 0x4
+	SEEK_SET                       = 0x0
 	SHUT_RD                        = 0x0
 	SHUT_RDWR                      = 0x2
 	SHUT_WR                        = 0x1
diff --git unix/zerrors_freebsd_arm64.go unix/zerrors_freebsd_arm64.go
index 218d399..4c83771 100644
--- unix/zerrors_freebsd_arm64.go
+++ unix/zerrors_freebsd_arm64.go
@@ -1298,6 +1298,11 @@ const (
 	SCM_RIGHTS                     = 0x1
 	SCM_TIMESTAMP                  = 0x2
 	SCM_TIME_INFO                  = 0x7
+	SEEK_CUR                       = 0x1
+	SEEK_DATA                      = 0x3
+	SEEK_END                       = 0x2
+	SEEK_HOLE                      = 0x4
+	SEEK_SET                       = 0x0
 	SHUT_RD                        = 0x0
 	SHUT_RDWR                      = 0x2
 	SHUT_WR                        = 0x1
diff --git unix/zerrors_linux_386.go unix/zerrors_linux_386.go
index 09fc559..562bef1 100644
--- unix/zerrors_linux_386.go
+++ unix/zerrors_linux_386.go
@@ -269,6 +269,12 @@ const (
 	SCM_TIMESTAMPNS                  = 0x23
 	SCM_TXTIME                       = 0x3d
 	SCM_WIFI_STATUS                  = 0x29
+	SEEK_CUR                         = 0x1
+	SEEK_DATA                        = 0x3
+	SEEK_END                         = 0x2
+	SEEK_HOLE                        = 0x4
+	SEEK_MAX                         = 0x4
+	SEEK_SET                         = 0x0
 	SFD_CLOEXEC                      = 0x80000
 	SFD_NONBLOCK                     = 0x800
 	SIOCATMARK                       = 0x8905
diff --git unix/zerrors_linux_amd64.go unix/zerrors_linux_amd64.go
index 75730cc..056b1b7 100644
--- unix/zerrors_linux_amd64.go
+++ unix/zerrors_linux_amd64.go
@@ -270,6 +270,12 @@ const (
 	SCM_TIMESTAMPNS                  = 0x23
 	SCM_TXTIME                       = 0x3d
 	SCM_WIFI_STATUS                  = 0x29
+	SEEK_CUR                         = 0x1
+	SEEK_DATA                        = 0x3
+	SEEK_END                         = 0x2
+	SEEK_HOLE                        = 0x4
+	SEEK_MAX                         = 0x4
+	SEEK_SET                         = 0x0
 	SFD_CLOEXEC                      = 0x80000
 	SFD_NONBLOCK                     = 0x800
 	SIOCATMARK                       = 0x8905
diff --git unix/zerrors_linux_arm.go unix/zerrors_linux_arm.go
index 127cf17..d09f9ca 100644
--- unix/zerrors_linux_arm.go
+++ unix/zerrors_linux_arm.go
@@ -276,6 +276,12 @@ const (
 	SCM_TIMESTAMPNS                  = 0x23
 	SCM_TXTIME                       = 0x3d
 	SCM_WIFI_STATUS                  = 0x29
+	SEEK_CUR                         = 0x1
+	SEEK_DATA                        = 0x3
+	SEEK_END                         = 0x2
+	SEEK_HOLE                        = 0x4
+	SEEK_MAX                         = 0x4
+	SEEK_SET                         = 0x0
 	SFD_CLOEXEC                      = 0x80000
 	SFD_NONBLOCK                     = 0x800
 	SIOCATMARK                       = 0x8905
diff --git unix/zerrors_linux_arm64.go unix/zerrors_linux_arm64.go
index 957ca1f..c077939 100644
--- unix/zerrors_linux_arm64.go
+++ unix/zerrors_linux_arm64.go
@@ -266,6 +266,12 @@ const (
 	SCM_TIMESTAMPNS                  = 0x23
 	SCM_TXTIME                       = 0x3d
 	SCM_WIFI_STATUS                  = 0x29
+	SEEK_CUR                         = 0x1
+	SEEK_DATA                        = 0x3
+	SEEK_END                         = 0x2
+	SEEK_HOLE                        = 0x4
+	SEEK_MAX                         = 0x4
+	SEEK_SET                         = 0x0
 	SFD_CLOEXEC                      = 0x80000
 	SFD_NONBLOCK                     = 0x800
 	SIOCATMARK                       = 0x8905
diff --git unix/zerrors_linux_mips.go unix/zerrors_linux_mips.go
index 314a205..a37447c 100644
--- unix/zerrors_linux_mips.go
+++ unix/zerrors_linux_mips.go
@@ -269,6 +269,12 @@ const (
 	SCM_TIMESTAMPNS                  = 0x23
 	SCM_TXTIME                       = 0x3d
 	SCM_WIFI_STATUS                  = 0x29
+	SEEK_CUR                         = 0x1
+	SEEK_DATA                        = 0x3
+	SEEK_END                         = 0x2
+	SEEK_HOLE                        = 0x4
+	SEEK_MAX                         = 0x4
+	SEEK_SET                         = 0x0
 	SFD_CLOEXEC                      = 0x80000
 	SFD_NONBLOCK                     = 0x80
 	SIOCATMARK                       = 0x40047307
diff --git unix/zerrors_linux_mips64.go unix/zerrors_linux_mips64.go
index 457e8de..96a623e 100644
--- unix/zerrors_linux_mips64.go
+++ unix/zerrors_linux_mips64.go
@@ -269,6 +269,12 @@ const (
 	SCM_TIMESTAMPNS                  = 0x23
 	SCM_TXTIME                       = 0x3d
 	SCM_WIFI_STATUS                  = 0x29
+	SEEK_CUR                         = 0x1
+	SEEK_DATA                        = 0x3
+	SEEK_END                         = 0x2
+	SEEK_HOLE                        = 0x4
+	SEEK_MAX                         = 0x4
+	SEEK_SET                         = 0x0
 	SFD_CLOEXEC                      = 0x80000
 	SFD_NONBLOCK                     = 0x80
 	SIOCATMARK                       = 0x40047307
diff --git unix/zerrors_linux_mips64le.go unix/zerrors_linux_mips64le.go
index 33cd28f..f652261 100644
--- unix/zerrors_linux_mips64le.go
+++ unix/zerrors_linux_mips64le.go
@@ -269,6 +269,12 @@ const (
 	SCM_TIMESTAMPNS                  = 0x23
 	SCM_TXTIME                       = 0x3d
 	SCM_WIFI_STATUS                  = 0x29
+	SEEK_CUR                         = 0x1
+	SEEK_DATA                        = 0x3
+	SEEK_END                         = 0x2
+	SEEK_HOLE                        = 0x4
+	SEEK_MAX                         = 0x4
+	SEEK_SET                         = 0x0
 	SFD_CLOEXEC                      = 0x80000
 	SFD_NONBLOCK                     = 0x80
 	SIOCATMARK                       = 0x40047307
diff --git unix/zerrors_linux_mipsle.go unix/zerrors_linux_mipsle.go
index 0e085ba..73cde7e 100644
--- unix/zerrors_linux_mipsle.go
+++ unix/zerrors_linux_mipsle.go
@@ -269,6 +269,12 @@ const (
 	SCM_TIMESTAMPNS                  = 0x23
 	SCM_TXTIME                       = 0x3d
 	SCM_WIFI_STATUS                  = 0x29
+	SEEK_CUR                         = 0x1
+	SEEK_DATA                        = 0x3
+	SEEK_END                         = 0x2
+	SEEK_HOLE                        = 0x4
+	SEEK_MAX                         = 0x4
+	SEEK_SET                         = 0x0
 	SFD_CLOEXEC                      = 0x80000
 	SFD_NONBLOCK                     = 0x80
 	SIOCATMARK                       = 0x40047307
diff --git unix/zerrors_linux_ppc.go unix/zerrors_linux_ppc.go
index 1b5928c..a49f557 100644
--- unix/zerrors_linux_ppc.go
+++ unix/zerrors_linux_ppc.go
@@ -324,6 +324,12 @@ const (
 	SCM_TIMESTAMPNS                  = 0x23
 	SCM_TXTIME                       = 0x3d
 	SCM_WIFI_STATUS                  = 0x29
+	SEEK_CUR                         = 0x1
+	SEEK_DATA                        = 0x3
+	SEEK_END                         = 0x2
+	SEEK_HOLE                        = 0x4
+	SEEK_MAX                         = 0x4
+	SEEK_SET                         = 0x0
 	SFD_CLOEXEC                      = 0x80000
 	SFD_NONBLOCK                     = 0x800
 	SIOCATMARK                       = 0x8905
diff --git unix/zerrors_linux_ppc64.go unix/zerrors_linux_ppc64.go
index f3a41d6..67a01b1 100644
--- unix/zerrors_linux_ppc64.go
+++ unix/zerrors_linux_ppc64.go
@@ -328,6 +328,12 @@ const (
 	SCM_TIMESTAMPNS                  = 0x23
 	SCM_TXTIME                       = 0x3d
 	SCM_WIFI_STATUS                  = 0x29
+	SEEK_CUR                         = 0x1
+	SEEK_DATA                        = 0x3
+	SEEK_END                         = 0x2
+	SEEK_HOLE                        = 0x4
+	SEEK_MAX                         = 0x4
+	SEEK_SET                         = 0x0
 	SFD_CLOEXEC                      = 0x80000
 	SFD_NONBLOCK                     = 0x800
 	SIOCATMARK                       = 0x8905
diff --git unix/zerrors_linux_ppc64le.go unix/zerrors_linux_ppc64le.go
index 6a5a555..c95ac09 100644
--- unix/zerrors_linux_ppc64le.go
+++ unix/zerrors_linux_ppc64le.go
@@ -328,6 +328,12 @@ const (
 	SCM_TIMESTAMPNS                  = 0x23
 	SCM_TXTIME                       = 0x3d
 	SCM_WIFI_STATUS                  = 0x29
+	SEEK_CUR                         = 0x1
+	SEEK_DATA                        = 0x3
+	SEEK_END                         = 0x2
+	SEEK_HOLE                        = 0x4
+	SEEK_MAX                         = 0x4
+	SEEK_SET                         = 0x0
 	SFD_CLOEXEC                      = 0x80000
 	SFD_NONBLOCK                     = 0x800
 	SIOCATMARK                       = 0x8905
diff --git unix/zerrors_linux_riscv64.go unix/zerrors_linux_riscv64.go
index a4da67e..97019b5 100644
--- unix/zerrors_linux_riscv64.go
+++ unix/zerrors_linux_riscv64.go
@@ -257,6 +257,12 @@ const (
 	SCM_TIMESTAMPNS                  = 0x23
 	SCM_TXTIME                       = 0x3d
 	SCM_WIFI_STATUS                  = 0x29
+	SEEK_CUR                         = 0x1
+	SEEK_DATA                        = 0x3
+	SEEK_END                         = 0x2
+	SEEK_HOLE                        = 0x4
+	SEEK_MAX                         = 0x4
+	SEEK_SET                         = 0x0
 	SFD_CLOEXEC                      = 0x80000
 	SFD_NONBLOCK                     = 0x800
 	SIOCATMARK                       = 0x8905
diff --git unix/zerrors_linux_s390x.go unix/zerrors_linux_s390x.go
index a7028e0..3a306b5 100644
--- unix/zerrors_linux_s390x.go
+++ unix/zerrors_linux_s390x.go
@@ -332,6 +332,12 @@ const (
 	SCM_TIMESTAMPNS                  = 0x23
 	SCM_TXTIME                       = 0x3d
 	SCM_WIFI_STATUS                  = 0x29
+	SEEK_CUR                         = 0x1
+	SEEK_DATA                        = 0x3
+	SEEK_END                         = 0x2
+	SEEK_HOLE                        = 0x4
+	SEEK_MAX                         = 0x4
+	SEEK_SET                         = 0x0
 	SFD_CLOEXEC                      = 0x80000
 	SFD_NONBLOCK                     = 0x800
 	SIOCATMARK                       = 0x8905
diff --git unix/zerrors_linux_sparc64.go unix/zerrors_linux_sparc64.go
index ed3b328..077f40b 100644
--- unix/zerrors_linux_sparc64.go
+++ unix/zerrors_linux_sparc64.go
@@ -323,6 +323,12 @@ const (
 	SCM_TIMESTAMPNS                  = 0x21
 	SCM_TXTIME                       = 0x3f
 	SCM_WIFI_STATUS                  = 0x25
+	SEEK_CUR                         = 0x1
+	SEEK_DATA                        = 0x3
+	SEEK_END                         = 0x2
+	SEEK_HOLE                        = 0x4
+	SEEK_MAX                         = 0x4
+	SEEK_SET                         = 0x0
 	SFD_CLOEXEC                      = 0x400000
 	SFD_NONBLOCK                     = 0x4000
 	SIOCATMARK                       = 0x8905
