diff --git language/go/generate.go language/go/generate.go
index 53e397a..38855c6 100644
--- language/go/generate.go
+++ language/go/generate.go
@@ -483,7 +483,7 @@ func (g *generator) generateLib(pkg *goPackage, embed string) *rule.Rule {
 	} else {
 		visibility = g.commonVisibility(pkg.importPath)
 	}
-	g.setCommonAttrs(goLibrary, pkg.rel, visibility, pkg.library, embed)
+	g.setCommonAttrs(goLibrary, pkg.rel, visibility, pkg.library, embed, true)
 	g.setImportAttrs(goLibrary, pkg.importPath)
 	return goLibrary
 }
@@ -512,7 +512,7 @@ func (g *generator) generateBin(pkg *goPackage, library string) *rule.Rule {
 		return goBinary // empty
 	}
 	visibility := g.commonVisibility(pkg.importPath)
-	g.setCommonAttrs(goBinary, pkg.rel, visibility, pkg.binary, library)
+	g.setCommonAttrs(goBinary, pkg.rel, visibility, pkg.binary, library, true)
 	return goBinary
 }
 
@@ -552,7 +552,7 @@ func (g *generator) generateTests(pkg *goPackage, library string) []*rule.Rule {
 		if test.hasInternalTest {
 			embed = library
 		}
-		g.setCommonAttrs(goTest, pkg.rel, nil, test, embed)
+		g.setCommonAttrs(goTest, pkg.rel, nil, test, embed, false)
 		if pkg.hasTestdata {
 			goTest.SetAttr("data", rule.GlobValue{Patterns: []string{"testdata/**"}})
 		}
@@ -629,9 +629,13 @@ func (g *generator) maybeGenerateExtraLib(lib *rule.Rule, pkg *goPackage) *rule.
 	return r
 }
 
-func (g *generator) setCommonAttrs(r *rule.Rule, pkgRel string, visibility []string, target goTarget, embed string) {
+func (g *generator) setCommonAttrs(r *rule.Rule, pkgRel string, visibility []string, target goTarget, embed string, flattenSrcs bool) {
 	if !target.sources.isEmpty() {
-		r.SetAttr("srcs", target.sources.buildFlat())
+		if flattenSrcs {
+			r.SetAttr("srcs", target.sources.buildFlat())
+		} else {
+			r.SetAttr("srcs", target.sources.build())
+		}
 	}
 	if !target.embedSrcs.isEmpty() {
 		r.SetAttr("embedsrcs", target.embedSrcs.build())
