diff --git internal/bzlmod/go_deps.bzl internal/bzlmod/go_deps.bzl
index 09f9af5..d508038 100644
--- internal/bzlmod/go_deps.bzl
+++ internal/bzlmod/go_deps.bzl
@@ -425,7 +425,7 @@ def _go_deps_impl(module_ctx):
 
             # Collect the relative path of the root module's go.mod file if it lives in the main
             # repository.
-            if module.is_root and not from_file_tag.go_mod.workspace_name:
+            if module.is_root and not from_file_tag.go_mod.repo_name:
                 go_mod = "go.mod"
                 if from_file_tag.go_mod.package:
                     go_mod = from_file_tag.go_mod.package + "/" + go_mod
@@ -453,7 +453,7 @@ def _go_deps_impl(module_ctx):
                 if module_path not in bazel_deps or version > bazel_deps[module_path].version:
                     bazel_deps[module_path] = struct(
                         module_name = module.name,
-                        repo_name = "@" + from_file_tag.go_mod.workspace_name,
+                        repo_name = "@" + from_file_tag.go_mod.repo_name,
                         version = version,
                         raw_version = raw_version,
                     )
diff --git internal/bzlmod/go_mod.bzl internal/bzlmod/go_mod.bzl
index 5baed71..30c2bd0 100644
--- internal/bzlmod/go_mod.bzl
+++ internal/bzlmod/go_mod.bzl
@@ -106,7 +106,7 @@ def parse_go_work(content, go_work_label):
 
     major, minor = go.split(".")[:2]
 
-    go_mods = [use_spec_to_label(go_work_label.workspace_name, use) for use in state["use"]]
+    go_mods = [use_spec_to_label(go_work_label.repo_name, use) for use in state["use"]]
     from_file_tags = [struct(go_mod = go_mod, _is_dev_dependency = False) for go_mod in go_mods]
 
     module_tags = [struct(version = mod.version, path = mod.to_path, _parent_label = go_work_label, local_path = mod.local_path, indirect = False) for mod in state["replace"].values()]
@@ -432,7 +432,7 @@ def parse_sumfile(module_ctx, label, sumfile):
     # changes. We have to use a canonical label as we may not have visibility
     # into the module that provides the sumfile
     sum_label = Label("@@{}//{}:{}".format(
-        label.workspace_name,
+        label.repo_name,
         label.package,
         sumfile,
     ))
diff --git internal/bzlmod/non_module_deps.bzl internal/bzlmod/non_module_deps.bzl
index d7572c2..b269a61 100644
--- internal/bzlmod/non_module_deps.bzl
+++ internal/bzlmod/non_module_deps.bzl
@@ -42,8 +42,8 @@ visibility("//")
 def _non_module_deps_impl(module_ctx):
     go_repository_cache(
         name = "bazel_gazelle_go_repository_cache",
-        # Label.workspace_name is always a canonical name, so use a canonical label.
-        go_sdk_name = "@" + HOST_COMPATIBLE_SDK.workspace_name,
+        # Label.repo_name is always a canonical name, so use a canonical label.
+        go_sdk_name = "@" + HOST_COMPATIBLE_SDK.repo_name,
         go_env = GO_ENV,
     )
     go_repository_tools(
