diff --git cmd/gazelle/fix.go cmd/gazelle/fix.go
index a58afc0..a5d9735 100644
--- cmd/gazelle/fix.go
+++ cmd/gazelle/fix.go
@@ -34,6 +34,7 @@ func fixFile(c *config.Config, f *rule.File) error {
 	if err := os.MkdirAll(filepath.Dir(outPath), 0o777); err != nil {
 		return err
 	}
+	os.Remove(outPath)
 	if err := os.WriteFile(outPath, newContent, 0o666); err != nil {
 		return err
 	}
diff --git rule/rule.go rule/rule.go
index b15f87f..862f95e 100644
--- rule/rule.go
+++ rule/rule.go
@@ -424,6 +424,7 @@ func (f *File) SortMacro() {
 func (f *File) Save(path string) error {
 	f.Sync()
 	f.Content = bzl.Format(f.File)
+	os.Remove(path)
 	return os.WriteFile(path, f.Content, 0o666)
 }
 
