diff --git lib/private/expand_variables.bzl lib/private/expand_variables.bzl
index a1f75bf..b352727 100644
--- lib/private/expand_variables.bzl
+++ lib/private/expand_variables.bzl
@@ -83,4 +83,35 @@ def expand_variables(ctx, s, outs = [], attribute_name = "args"):
     )
     additional_substitutions["WORKSPACE"] = ctx.workspace_name
 
-    return ctx.expand_make_variables(attribute_name, s, additional_substitutions)
+    def get_value(variable_name):
+        if variable_name in additional_substitutions:
+            return additional_substitutions[variable_name]
+        return ctx.var[variable_name]
+
+    result = ""
+    state = 0
+    variable_name = ""
+    for c in s.elems():
+        if state == 0:
+            if c == "$":
+                state = 1
+            else:
+                result += c
+        elif state == 1:
+            if c == "(":
+                state = 2
+            elif c == "@" or c == "<" or c == "^":
+                result += get_value(c)
+            else:
+                fail("unknown sequence $%s" % c)
+        elif state == 2:
+            if c == ")":
+                result += get_value(variable_name)
+                variable_name = ""
+                state = 0
+            else:
+                variable_name += c
+        else:
+            fail("bad state")
+
+    return result
