--- gen-go/agent/metrics/v1/BUILD.bazel.orig	2019-04-30 11:03:46.000000000 -0700
+++ gen-go/agent/metrics/v1/BUILD.bazel	2019-04-30 11:05:35.000000000 -0700
@@ -6,6 +6,9 @@
     importpath = "github.com/census-instrumentation/opencensus-proto/gen-go/agent/metrics/v1",
     visibility = ["//visibility:public"],
     deps = [
+        "//gen-go/agent/common/v1:go_default_library",
+        "//gen-go/metrics/v1:go_default_library",
+        "//gen-go/resource/v1:go_default_library",
         "@com_github_golang_protobuf//proto:go_default_library",
         "@org_golang_google_grpc//:go_default_library",
         "@org_golang_google_grpc//codes:go_default_library",
--- gen-go/metrics/v1//BUILD.bazel.orig	2019-04-30 10:40:37.000000000 -0700
+++ gen-go/metrics/v1//BUILD.bazel	2019-04-30 10:51:32.000000000 -0700
@@ -6,6 +6,7 @@
     importpath = "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1",
     visibility = ["//visibility:public"],
     deps = [
+	"//gen-go/resource/v1:go_default_library",
         "@com_github_golang_protobuf//proto:go_default_library",
         "@io_bazel_rules_go//proto/wkt:timestamp_go_proto",
         "@io_bazel_rules_go//proto/wkt:wrappers_go_proto",
--- gen-go/trace/v1/BUILD.bazel.orig	2019-04-30 10:52:41.000000000 -0700
+++ gen-go/trace/v1/BUILD.bazel	2019-04-30 10:52:50.000000000 -0700
@@ -9,6 +9,7 @@
     importpath = "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1",
     visibility = ["//visibility:public"],
     deps = [
+	"//gen-go/resource/v1:go_default_library",
         "@com_github_golang_protobuf//proto:go_default_library",
         "@io_bazel_rules_go//proto/wkt:timestamp_go_proto",
         "@io_bazel_rules_go//proto/wkt:wrappers_go_proto",
--- gen-go/agent/trace/v1/BUILD.bazel.orig	2019-04-30 11:15:32.000000000 -0700
+++ gen-go/agent/trace/v1/BUILD.bazel	2019-04-30 11:16:39.000000000 -0700
@@ -9,6 +9,10 @@
     importpath = "github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1",
     visibility = ["//visibility:public"],
     deps = [
+        "//gen-go/agent/common/v1:go_default_library",
+        "//gen-go/metrics/v1:go_default_library",
+        "//gen-go/resource/v1:go_default_library",
+        "//gen-go/trace/v1:go_default_library",
         "@com_github_golang_protobuf//proto:go_default_library",
         "@com_github_grpc_ecosystem_grpc_gateway//runtime:go_default_library",
         "@com_github_grpc_ecosystem_grpc_gateway//utilities:go_default_library",
